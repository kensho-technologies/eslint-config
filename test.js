/* eslint-env jest */
const path = require('path')

const {CLIEngine} = require('eslint')

const baseConfig = require('.')

test('lints all fixtures', () => {
  const cli = new CLIEngine({
    baseConfig,
    extensions: ['.js', '.ts', '.tsx'],
    ignore: false,
    useEslintrc: false,
  })
  const {results} = cli.executeOnFiles([`${__dirname}/fixtures`])
  results.forEach((result) => {
    const basename = path.basename(result.filePath)
    const messages = result.messages.map((message) => ({
      line: message.line,
      column: message.column,
      message: message.message,
      rule: message.ruleId,
      severity: message.severity,
    }))
    expect(messages).toMatchSnapshot(basename)
  })
})
