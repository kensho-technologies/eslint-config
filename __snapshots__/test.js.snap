// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lints all fixtures: class.tsx 1`] = `[]`;

exports[`lints all fixtures: component.js 1`] = `
[
  {
    "column": 7,
    "line": 72,
    "message": "Prefer fragment shorthand over React.Fragment",
    "rule": "react/jsx-fragments",
    "severity": 2,
  },
]
`;

exports[`lints all fixtures: emotion.tsx 1`] = `
[
  {
    "column": 1,
    "line": 1,
    "message": "'@emotion/react' should be listed in the project's dependencies, not devDependencies.",
    "rule": "import/no-extraneous-dependencies",
    "severity": 2,
  },
]
`;

exports[`lints all fixtures: experimental.tsx 1`] = `[]`;

exports[`lints all fixtures: functionComponent.tsx 1`] = `
[
  {
    "column": 1,
    "line": 1,
    "message": "'react' should be listed in the project's dependencies, not devDependencies.",
    "rule": "import/no-extraneous-dependencies",
    "severity": 2,
  },
  {
    "column": 25,
    "line": 14,
    "message": "Must use destructuring theme assignment",
    "rule": "react/destructuring-assignment",
    "severity": 2,
  },
]
`;

exports[`lints all fixtures: import.js 1`] = `
[
  {
    "column": 25,
    "line": 1,
    "message": "baz not found in './js-exports'",
    "rule": "import/named",
    "severity": 2,
  },
  {
    "column": 27,
    "line": 2,
    "message": "Missing file extension for "./nonexistent"",
    "rule": "import/extensions",
    "severity": 2,
  },
  {
    "column": 27,
    "line": 2,
    "message": "Unable to resolve path to module './nonexistent'.",
    "rule": "import/no-unresolved",
    "severity": 2,
  },
]
`;

exports[`lints all fixtures: import.ts 1`] = `
[
  {
    "column": 27,
    "line": 2,
    "message": "Missing file extension for "./nonexistent"",
    "rule": "import/extensions",
    "severity": 2,
  },
  {
    "column": 27,
    "line": 2,
    "message": "Unable to resolve path to module './nonexistent'.",
    "rule": "import/no-unresolved",
    "severity": 2,
  },
]
`;

exports[`lints all fixtures: import-ts-from-js.js 1`] = `[]`;

exports[`lints all fixtures: js-exports.js 1`] = `[]`;

exports[`lints all fixtures: jsdoc.ts 1`] = `[]`;

exports[`lints all fixtures: prop-types.js 1`] = `
[
  {
    "column": 5,
    "line": 9,
    "message": "'propWithoutPropType' is missing in props validation",
    "rule": "react/prop-types",
    "severity": 2,
  },
  {
    "column": 3,
    "line": 27,
    "message": "'unusedProp' PropType is defined but prop is never used",
    "rule": "react/no-unused-prop-types",
    "severity": 2,
  },
  {
    "column": 3,
    "line": 32,
    "message": "defaultProp "requiredPropWithDefault" defined for isRequired propType.",
    "rule": "react/default-props-match-prop-types",
    "severity": 2,
  },
]
`;

exports[`lints all fixtures: prop-types.tsx 1`] = `
[
  {
    "column": 16,
    "line": 3,
    "message": "Missing return type on function.",
    "rule": "@typescript-eslint/explicit-function-return-type",
    "severity": 2,
  },
  {
    "column": 3,
    "line": 32,
    "message": "defaultProp "requiredPropWithDefault" defined for isRequired propType.",
    "rule": "react/default-props-match-prop-types",
    "severity": 2,
  },
]
`;

exports[`lints all fixtures: ts-exports.ts 1`] = `[]`;

exports[`lints all fixtures: variables.js 1`] = `
[
  {
    "column": 24,
    "line": 1,
    "message": "Unexpected use of 'find'.",
    "rule": "no-restricted-globals",
    "severity": 2,
  },
  {
    "column": 24,
    "line": 2,
    "message": "'implicitGlobal' is not defined.",
    "rule": "no-undef",
    "severity": 2,
  },
  {
    "column": 7,
    "line": 3,
    "message": "'unusedA' is assigned a value but never used.",
    "rule": "no-unused-vars",
    "severity": 2,
  },
  {
    "column": 33,
    "line": 6,
    "message": "Unexpected block statement surrounding arrow body; move the returned value immediately after the \`=>\`.",
    "rule": "arrow-body-style",
    "severity": 2,
  },
]
`;

exports[`lints all fixtures: variables.ts 1`] = `
[
  {
    "column": 24,
    "line": 1,
    "message": "Unexpected use of 'find'.",
    "rule": "no-restricted-globals",
    "severity": 2,
  },
  {
    "column": 26,
    "line": 6,
    "message": "Missing return type on function.",
    "rule": "@typescript-eslint/explicit-function-return-type",
    "severity": 2,
  },
  {
    "column": 33,
    "line": 6,
    "message": "Unexpected block statement surrounding arrow body; move the returned value immediately after the \`=>\`.",
    "rule": "arrow-body-style",
    "severity": 2,
  },
  {
    "column": 14,
    "line": 12,
    "message": "Variable name \`foo_encouraged\` must match one of the following formats: camelCase, UPPER_CASE, PascalCase",
    "rule": "@typescript-eslint/naming-convention",
    "severity": 2,
  },
]
`;
